src/greenlight/test.clj

```diff
@@ -126,12 +126,33 @@
   nil)
 
 
+(defn- prompt-for-retry
+  "Enter a retry prompt loop. Return true if retry selected, false otherwise."
+  []
+  (loop []
+    (print (str "Retry? [y/n] "))
+    (flush)
+    (let [response (read-line)]
+      (cond
+        (re-find #"(?i)^\s*y(?:es)?\s*$" response) true
+        (re-find #"(?i)^\s*no?\s*$" response) false
+        :else (recur)))))
```

Comment:
This could be simplified and doesn't need regular expressions:
```suggestion
    (case (str/lower-case (str/trim (read-line)))]
      ("y" "yes") true
      ("n" "no") false
      (recur))
```